cprover_bindings/src/goto_program/location.rs

```diff
@@ -23,6 +23,8 @@ pub enum Location {
         comment: InternedString,
         property_class: InternedString,
     },
+    /// Covers cases where there are no Location Details but Property Class is needed
+    NoneProperty { comment: InternedString, property_class: InternedString },
```

Comment:
Perhaps rename to `PropertyUnknownLocation`?

Do we know the scenarios for which the location is missing? Sounds like something that needs to be fixed, in which case this new variant won't be needed.

Reply:
Sometimes we call `location::none()` and pass it to the assert deliberately , one example of this is wherever we call `codegen_unimplemented()`. We always set the location as None for such cases and i thought adding this new variant would allow us to retain that behavior while setting property_class and description.